plugins/api/front.js

```diff
@@ -0,0 +1,26 @@
+const { ipcRenderer } = require('electron');
+
+module.exports = () => {
+    let currentStatus
+
+    function getLikeStatus(currentVideoId = document.querySelector('#movie_player')?.getPlayerResponse()?.videoDetails?.videoId) {
+        const playerBar = document.querySelector("ytmusic-player-bar");
+        const likedVideos = playerBar?.getState()?.likeStatus?.videos;
+        if (likedVideos) {
+            for (const [id, status] of Object.entries(likedVideos)) {
+                if (id === currentVideoId) {
+                    return status;
+                }
+            }
+        }
+        return playerBar?.__data?.likeButtonRenderer_?.likeStatus;
+    }
+
+    const loop = () => {
+        if (currentStatus !== getLikeStatus()) ipcRenderer.send('api-like-button-status', getLikeStatus())
+        currentStatus = getLikeStatus()
+        setTimeout(loop, 100)
+    }
+
+    loop()
```

Comment:
```suggestion
const likeObserver = new MutationObserver((mutations) => {
	mutations.forEach((mutation) => {
		ipcRenderer.send(
			"api-like-button-status",
			mutation.target.getAttribute("like-status"),
		);
	});
});

likeObserver.observe(document.querySelector("ytmusic-like-button-renderer"), {
	attributes: true,
	childList: false,
	characterData: false,
	subtree: false,
	attributeFilter: ["like-status"],
});
```

Reply:
I didn't know you intended to poll it when you asked about it… this should be much better than a loop that runs 10 times per second

Reply:
Alright thanks.